src/search.cpp

```diff
@@ -209,12 +212,15 @@ void GreedySearch_Cpu::SampleTopKTopP(int k, float p, float temperature) {
     std::vector<int> indices(scores.size());
     std::iota(indices.begin(), indices.end(), 0);
     std::partial_sort(indices.begin(), indices.begin() + k, indices.end(), [scores = scores.data()](int i, int j) { return scores[i] > scores[j]; });
+    std::vector<float> top_k_scores(k);
+    for (int i = 0; i < k; i++)
+      top_k_scores.push_back(scores[indices[i]]);
```

Comment:
The original code was correct here, as it's doing threshold -= scores[indices[i]].

The new code is causing an access violation, since indices[i] will be larger than top_k_scores size. Just revert the change to fix it.